packages/loader/container-loader/src/loader.ts

```diff
@@ -489,6 +489,7 @@ export class Loader implements IHostLoader {
                 loadMode: request.headers?.[LoaderHeader.loadMode],
             },
             pendingLocalState,
+            request.headers?.[LoaderHeader.clientDetails]?.environment?.startsWith("test-"), // allowLegacy
```

Comment:
That's too implicit - it will take some effort for someone unrelated to find what code is actually dependent on it. Is there any way to make it more explicit? Like maybe environment to have some unique keyword that we look for (not ideal, but a bit better than just test-). If there is way to make it even more explicit and searchable, we should do it

Reply:
I'll reconsider this